cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java

```diff
@@ -177,17 +176,21 @@ public void delete(String name) throws IOException {
   /**
    * List of all the entries in the secure store.
    * @return A list of {@link SecureStoreMetadata} objects representing the data stored in the store.
+   * @param namespace The namespace this key belongs to.
    */
   @Override
-  public List<SecureStoreMetadata> list() throws IOException {
+  public List<SecureStoreMetadata> list(String namespace) throws IOException {
     readLock.lock();
     try {
       Enumeration<String> aliases = keyStore.aliases();
-      String name;
       List<SecureStoreMetadata> list = new ArrayList<>();
+      String prefix = namespace + nameSeparator;
       while (aliases.hasMoreElements()) {
-        name = aliases.nextElement();
-        list.add(getSecureStoreMetadata(name));
+        String alias = aliases.nextElement();
+        // Filter out elements not in this namespace.
+        if (alias.startsWith(prefix)) {
```

Comment:
irrespective of this check though, validation should happen also because the name of the key appears in the REST API path. Perhaps use something like ``EntityId.isId`` to validate the key in the REST API.